pkgs/by-name/al/alsa-utils/package.nix

```diff
@@ -1,5 +1,17 @@
-{lib, stdenv, fetchurl, fetchpatch, alsa-lib, gettext, makeWrapper, ncurses, libsamplerate, pciutils, which, fftw}:
-
+{ lib, stdenv, fetchurl, fetchpatch, alsa-lib, alsa-plugins, gettext, makeWrapper, ncurses, libsamplerate, pciutils, which, fftw, pipewire, symlinkJoin }:
+
+let
```

Comment:
Does not this code makes  ALSA  dependent on Pipewire, when they are not really dependent? It should be optional and configurable somehow.

Reply:
I could certainly do something like:

```diff
@@ -1,7 +1,7 @@
 let
-  plugin-packages = [ alsa-plugins pipewire.lib ];
+  plugin-packages = [ alsa-plugins ] ++ lib.optional (pipewire != null) pipewire.lib;
 
   # Create a directory containing symlinks of all ALSA plugins.
```

Would that satisfy your needs?

What should be the default value? On one hand I understand that people shouldn't get unwanted dependencies. On the other hand, not including pipewire by default means that users of pipewire systems will get problems. FYI, we are talking about an additional 12MB for pipewire.lib.

Reply:
> Would that satisfy your needs?

Testing against null is discouraged - it is a source of heisenbugs, [as happened in Emacs a long time ago](https://github.com/NixOS/nixpkgs/pull/233301#issuecomment-1567661913).

The recommended approach is to use a Boolean guard, something like  `withPipewireLib ? true`.

Besides, OK to me.

> What should be the default value?

True by default is fine, since Pipewire is stable ans not too huge.

Reply:
@AndersonTorres done. Thank you for the guidance!